blueprints/cloud-operations/network-dashboard/cloud-function/metrics/routes.py

```diff
@@ -169,11 +175,121 @@ def get_dynamic_routes_ppg(config, metric_dict, usage_dict, limit_dict):
         peered_network_dict["usage"] = peered_usage
         peered_network_dict["limit"] = peered_limit
 
-      limits.count_effective_limit(config, project, network_dict,
+      limits.count_effective_limit(config, project_id, network_dict,
                                    metric_dict["usage"]["name"],
                                    metric_dict["limit"]["name"],
                                    metric_dict["utilization"]["name"],
-                                   limit_dict)
+                                   limit_dict, timestamp)
+      print(
+          f"Buffered {metric_dict['usage']['name']} for peering group {network_dict['network_name']} in {project_id}"
+      )
+
+
+def get_static_routes_dict(config):
+  '''
+    Calls the Asset Inventory API to get all static customr routes under the GCP organization.
+    Parameters:
+      config (dict): The dict containing config like clients and limits
+    Returns:
+      routes_per_vpc_dict (dictionary of string: int): Keys are the network links and values are the number of custom static routes per network.
+  '''
+  routes_per_vpc_dict = defaultdict()
+  usage_dict = defaultdict()
+
+  read_mask = field_mask_pb2.FieldMask()
+  read_mask.FromJsonString('name,versionedResources')
+
+  response = config["clients"]["asset_client"].search_all_resources(
+      request={
+          "scope": f"organizations/{config['organization']}",
+          "asset_types": ["compute.googleapis.com/Route"],
+          "read_mask": read_mask
+      })
+
+  for resource in response:
+    for versioned in resource.versioned_resources:
+      static_route = dict()
+      for field_name, field_value in versioned.resource.items():
+        static_route[field_name] = field_value
+      static_route["project_id"] = re.search("\/([^\/]*)(\/[^\/]*){3}$",
```

Comment:
I prefer using splits like field_value.split('/')[-1] to get the network name and field_value.split('/')[6] (to be checked) to get the project_id, I think it is more readable then regexp. What do you think?

Reply:
k

Reply:
k